### PR TITLE
Reduce CI time

### PR DESCRIPTION
**Pull request by kateinoigakukun**
- Cache with sccache
- Stop testing on Linux temporarily

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
This seems to be working pretty well, I wonder if testing on Linux could be reenabled again?

**Comment by kateinoigakukun** [comment]
Thanks! Even if CI time is saved, testing on Linux reaches the limitation of disk 😢

**Comment by MaxDesiatov** [comment]
I wonder if we could get a disk usage report with `du` or something similar and then cleanup the biggest unused files before the tests start running 🤔

**Comment by kateinoigakukun** [comment]
In my local environment, the biggest artifact is testing directory 🤔 I think the reason is that ICU data is linked statically into each testing executable file.

```
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/man
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/docs
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/runtimes
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/test
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/userdocs
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/validation-test-linux-x86_64
12K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/validation-test-wasi-wasm32
16K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/benchmarks
16K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/cmake
20K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/Testing
20K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/testdir
24K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/test
32K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/libexec
32K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/projects
32K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/docs
36K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/test-linux-x86_64
52K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/utils
88K     ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/.cmake
128K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/share
132K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/cmake
164K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/api_test
208K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/CMakeFiles
468K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/CMakeFiles
492K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/examples
904K    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/unittests
1.2M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/CMakeFiles
2.4M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/include
2.5M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/include
2.6M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64/src
3.2M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/cmark-linux-x86_64
4.8M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/tools
5.3M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/.cmake
5.7M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/swift-test-results
8.4M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/utils
19M     ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/.cmake
61M     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/module-cache
119M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/stdlib
238M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/unittests
292M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/tools
725M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/lib
831M    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/lib
1.6G    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64/bin
1.6G    ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/bin
2.8G    ./Ninja-ReleaseAssert+stdlib-DebugAssert/llvm-linux-x86_64
30G     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64/test-wasi-wasm32
33G     ./Ninja-ReleaseAssert+stdlib-DebugAssert/swift-linux-x86_64
36G     ./Ninja-ReleaseAssert+stdlib-DebugAssert
```

**Comment by MaxDesiatov** [comment]
Would it be hard then to optionally delete a test after it runs if they are compiled and executed one by one (I hope)?

**Comment by kateinoigakukun** [comment]
@MaxDesiatov Ummm, I think that way is quite hard. I think reducing binary size has more worth to do in the future.
Minimizing binary size is essential work we have to do.

File: utils/webassembly/linux/install-dependencies.sh [hunk 1]
```diff
@@ -46,3 +46,10 @@ ln -s wasm32-wasi wasi-sdk/share/wasi-sysroot/lib/wasm32-wasi-unknown
 
 wget -O icu.tar.xz "https://github.com/swiftwasm/icu4c-wasi/releases/download/0.3.0/icu4c-wasi.tar.xz"
 tar xf icu.tar.xz
+
+# Install sccache
+
+sudo mkdir /opt/sccache && cd /opt/sccache
+wget -O - "https://github.com/mozilla/sccache/releases/download/0.2.12/sccache-0.2.12-x86_64-unknown-linux-musl.tar.gz" | \
```
Comment (MaxDesiatov) [inline utils/webassembly/linux/install-dependencies.sh]:
```suggestion
wget -O - "https://github.com/mozilla/sccache/releases/download/0.2.13/sccache-0.2.13-x86_64-unknown-linux-musl.tar.gz" | \
```

File: utils/webassembly/macos/install-dependencies.sh [hunk 2]
```diff
@@ -5,9 +5,8 @@ set -ex
 brew uninstall python@2 || true
 brew install cmake ninja llvm
```
Comment (MaxDesiatov) [inline utils/webassembly/macos/install-dependencies.sh]:
```suggestion
brew install cmake ninja llvm sccache
```

**Pull request merged**